apis/nodecore/v1alpha1/common.go

```diff
@@ -43,13 +43,28 @@ type GenericRef struct {
 	// The namespace containing the resource to be referenced. It should be left
 	// empty in case of cluster-wide resources.
 	Namespace string `json:"namespace,omitempty"`
+
+	// The API version of the resource to be referenced.
+	APIVersion string `json:"apiVersion,omitempty"`
+
+	// The kind of the resource to be referenced.
+	Kind string `json:"kind,omitempty"`
+}
+
+func (g *GenericRef) String() string {
+	return fmt.Sprintf("%s:%s - %s/%s", g.APIVersion, g.Kind, g.Namespace, g.Name)
 }
 
 // NodeIdentity is the identity of a FLUIDOS Node.
 type NodeIdentity struct {
-	Domain string `json:"domain"`
-	NodeID string `json:"nodeID"`
-	IP     string `json:"ip"`
+	Domain                string                      `json:"domain"`
+	NodeID                string                      `json:"nodeID"`
+	IP                    string                      `json:"ip"`
+	AdditionalInformation *NodeIdentityAdditionalInfo `json:"additionalInformation,omitempty"`
```

Comment:
Is this `AdditionalInformation` related only to Liqo?

Reply:
The AdditionalInformation is a field that at the moment contain only the LiqoID information useful for other steps, but in the future can be extended.